src/services/expenses.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+let expenses = [];
+
+function getAll({ userId, categories, from, to }) {
+  if (userId) {
+    expenses = expenses.filter(e => e.id === Number(userId));
+  }
+
+  if (categories) {
+    expenses = expenses.filter(e => e.category === categories);
+  }
+
+  if (from && to) {
+    expenses = expenses.filter(
+      e => new Date(e.spentAt).valueOf() >= new Date(from).valueOf()
+        && new Date(e.spentAt).valueOf() <= new Date(to).valueOf()
+    );
+  }
+
+  return expenses;
+}
+
+function getById(expensesId) {
+  const foundExpenses = expenses.find(
+    expense => expense.id === expensesId
+  );
+
+  return foundExpenses || null;
+}
+
+function create(userId, spentAt, title, amount, category, note) {
+  const newExpenses = {
```

Comment:
const newExpense = {
    id: expenses.length + 1, // exp.length === last element in array + 1
    ...expense,
  };
  
  JUST FYI, you can write the same shorter

Reply:
Good job!

Reply:
true

Reply:
Sure but it's more readable for me